app/Http/Controllers/AlbumController.php

```diff
@@ -165,6 +165,108 @@ public function get(Request $request)
 		return $return;
 	}
 
+	/**
+	 * Provided an albumID, returns the album with only map related data.
+	 *
+	 * @param Request $request
+	 *
+	 * @return array|string
+	 */
+	public function getPositionData(Request $request)
+	{
+		$request->validate(['albumID' => 'string|required']);
+		$return = array();
+		$return['albums'] = array();
+		// Get photos
+		// Get album information
+		$UserId = $this->sessionFunctions->id();
+		$full_photo = Configs::get_value('full_photo', '1') == '1';
+
+		switch ($request['albumID']) {
+			case 'f':
+				if ($this->sessionFunctions->is_logged_in()) {
+					$user = User::find($UserId);
+
+					if ($UserId == 0 || $user->upload) {
+						$photos_sql = Photo::select_stars(Photo::OwnedBy($UserId));
+						break;
+					}
+				}
+				$photos_sql = Photo::select_stars(Photo::whereIn('album_id', $this->albumFunctions->getPublicAlbums()));
+				break;
+			case 's':
+				$photos_sql = Photo::select_public(Photo::OwnedBy($UserId));
+				break;
+			case 'r':
+				if ($this->sessionFunctions->is_logged_in()) {
+					$user = User::find($UserId);
+
+					if ($UserId == 0 || $user->upload) {
+						$photos_sql = Photo::select_recent(Photo::OwnedBy($UserId));
+						break;
+					}
+				}
+				$photos_sql = Photo::select_recent(Photo::whereIn('album_id', $this->albumFunctions->getPublicAlbums()));
+				break;
+			case '0':
+				$photos_sql = Photo::select_unsorted(Photo::OwnedBy($UserId));
+				break;
+			default:
+				$album = Album::with('children')->find($request['albumID']);
+				if ($album === null) {
+					Logs::error(__METHOD__, __LINE__, 'Could not find specified album');
+
+					return 'false';
+				}
+				$return = $album->prepareData();
+				// we just require is_logged_in for this one.
+				$username = null;
+				if ($this->sessionFunctions->is_logged_in()) {
+					$username = $album->owner->username;
+				}
+
+				$full_photo = $album->full_photo_visible();
+
+				$album_list = array();
+				if ($request['includeSubAlbums']) {
+					// Get all subalbums of the current album
+					$album_list = $this->albumFunctions->getAlbumIDsfromAlbumTree($this->albumFunctions->get_albums($album, $username));
```

Comment:
Shouldn't you be using `AlbumFunctions::get_sub_albums()` here? What you are doing seems more complicated than it needs to be, and more costly.